src/StreamJsonRpc.Shared/JsonRpc.cs

```diff
@@ -320,6 +346,37 @@ public Task InvokeWithCancellationAsync(string targetName, IReadOnlyList<object>
         }
 
         /// <summary>
+        /// Invoke a method on the server and get back the result.  The parameter is passed as an object.
+        /// </summary>
+        /// <typeparam name="Result">Type of the method result</typeparam>
+        /// <param name="targetName">The name of the method to invoke on the server. Must not be null or empty string.</param>
+        /// <param name="argument">Method arguments, must be serializable to JSON.</param>
+        /// <param name="cancellationToken">The token whose cancellation should signal the server to stop processing this request.</param>
+        /// <returns>A task that completes when the server method executes and returns the result.</returns>
+        /// <exception cref="OperationCanceledException">
+        /// Result task fails with this exception if the communication channel ends before the result gets back from the server
+        /// or in response to the <paramref name="cancellationToken"/> being canceled.
+        /// </exception>
+        /// <exception cref="RemoteInvocationException">
+        /// Result task fails with this exception if the server method throws an exception,
+        /// which may occur in response to the <paramref name="cancellationToken"/> being canceled.
+        /// </exception>
+        /// <exception cref="RemoteMethodNotFoundException">
+        /// Result task fails with this exception if the <paramref name="targetName"/> method is not found on the target object on the server.
+        /// </exception>
+        /// <exception cref="RemoteTargetNotSetException">
+        /// Result task fails with this exception if the server has no target object.
+        /// </exception>
+        /// <exception cref="ArgumentNullException">If <paramref name="targetName"/> is null.</exception>
+        /// <exception cref="ObjectDisposedException">If this instance of <see cref="JsonRpc"/> has been disposed.</exception>
+        public Task<Result> InvokeWithParameterPassedAsObjectWithCancellationAsync<Result>(string targetName, object argument = null, CancellationToken cancellationToken = default(CancellationToken))
```

Comment:
I don't think we need the two methods (one with CancellationToken and one without). Let's just have one (that takes an optional CancellationToken), and the shorter method name.

With the parameters array methods, we had to have it because the args came in as a params array, so we couldn't accept a CancellationToken parameter after that. But here, that's not a restriction.